plugins/@grouparoo/pipedrive/src/lib/export/exportProfile.ts

```diff
@@ -0,0 +1,128 @@
+import { ExportProfilePluginMethod } from "@grouparoo/core";
+import { connect } from "../connect";
+import {
+  PipedriveCacheData,
+  getKnownPersonFieldMap,
+} from "./destinationMappingOptions";
+import { getGroupFieldKey } from "./listMethods";
+
+export const exportProfile: ExportProfilePluginMethod = async ({
+  appId,
+  appOptions,
+  export: {
+    newProfileProperties,
+    oldProfileProperties,
+    newGroups,
+    oldGroups,
+    toDelete,
+  },
+}) => {
+  if (Object.keys(newProfileProperties).length === 0) {
+    return { success: true };
+  }
+
+  const client = await connect(appOptions);
+  const cacheData: PipedriveCacheData = { appId, appOptions };
+
+  const oldEmail = oldProfileProperties["Email"];
+  const foundId = await client.EnhancedPersonsController.findPersonIdByEmail(
+    oldEmail
+  );
```

Comment:
What happens if this is the first time we sync and want to delete?  Or, what if there was an email change (old -> new) and a toDelete?  

sendgrid does this:
```js
const user = await client.getUser(email);
  let oldUser = null;
  if (oldEmail && oldEmail !== email) {
    oldUser = await client.getUser(oldEmail);
  }

  if (toDelete) {
    const userToDelete = user || oldUser;
    if (userToDelete) {
      await client.deleteUsers([userToDelete["id"]]);
    }
    return { success: true };
  }
```

Reply:
You're right! Added some tests for this.

However, I believe sendgrid might have an issue... It seems like it should be `const userToDelete = oldUser || user`, no? If we're updating the email and deleting at the same time, the update would not have occurred so we need to delete the _old_ email, not the new. Correct me if I'm wrong please.

Also, I noticed sendgrid silently ignores deleting nonexistent users. I'm throwing an error in my case so that grouparoo retries if need be- what should be the right way to do it?

Reply:
I agree that you should be deleting the `oldEmail` in this case.  I think.  It's confusing :/

Whether or not to fail or silently when a user can't be found is a really good question, and one we need to address more globally with Grouparoo.  We are going to start making a distinction between `syncing data` (match my group exactly and delete people) and `enriching` data (update the people who are there, if you can, and don't delete them) which will make this more clear... hopefully.  Most destinations to-date are in the "sync" mode - which means we should delete people if we can. 

More to come!